modules/cloud-function/variables.tf

```diff
@@ -17,8 +17,8 @@
 variable "bucket_config" {
   description = "Enable and configure auto-created bucket. Set fields to null to use defaults."
   type = object({
-    location             = string
-    lifecycle_delete_age = number
+    location             = optional(string)
+    lifecycle_delete_age = optional(number)
   })
   default = null
```

Comment:
I think this variable should be non-nullable. Your location check depends on the object existing and having the two fields defined.
```hcl
  location = (
    var.bucket_config.location == null
    ? var.region
    : var.bucket_config.location
  )
```

Reply:
Existence of `bucket_config` is guaranteed few lines above with:
```
count                       = var.bucket_config == null ? 0 : 1
```

I like the current interface of the module, that if you don't provide `location` it will default `var.region`